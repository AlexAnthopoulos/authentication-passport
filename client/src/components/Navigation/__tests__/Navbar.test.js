import React from "react";
import Navbar from "../Navbar";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Link } from "react-router-dom";
import { StaticRouter } from "react-router";

const shallowComponent = () => shallow(<Navbar />);
const wrapper = shallowComponent();

test("shallow render", () => {
  expect(wrapper).toMatchSnapshot();
});

test("Link matches home snapshot", () => {
  const context = {};
  const component = renderer.create(
    <StaticRouter location="/" context={context}>
      <Link to="/" />
    </StaticRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Link matches login snapshot", () => {
  const context = {};
  const component = renderer.create(
    <StaticRouter location="/login" context={context}>
      <Link to="/login" />
    </StaticRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Link matches loprofilegin snapshot", () => {
  const context = {};
  const component = renderer.create(
    <StaticRouter location="/profile" context={context}>
      <Link to="/profile" />
    </StaticRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
