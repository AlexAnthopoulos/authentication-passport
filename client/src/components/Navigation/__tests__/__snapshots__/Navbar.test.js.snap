// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link matches home snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
/>
`;

exports[`Link matches login snapshot 1`] = `
<a
  href="/login"
  onClick={[Function]}
/>
`;

exports[`Link matches loprofilegin snapshot 1`] = `
<a
  href="/profile"
  onClick={[Function]}
/>
`;

exports[`Link matches profile snapshot 1`] = `
<a
  href="/profile"
  onClick={[Function]}
/>
`;

exports[`shallow render 1`] = `
<nav>
  <ul>
    <li>
      <Link
        to="/"
      >
        Register
      </Link>
    </li>
    <li>
      <Link
        to="/login"
      >
        Log In
      </Link>
    </li>
    <li>
      <Link
        to="/profile"
      >
        Profile
      </Link>
    </li>
  </ul>
</nav>
`;
